pandas/core/categorical.py

```diff
@@ -1085,10 +1085,14 @@ def searchsorted(self, v, side='left', sorter=None):
                              "ordered one")
 
         from pandas.core.series import Series
-        values_as_codes = self.categories.values.searchsorted(
-            Series(v).values, side=side)
+        values_as_codes = _get_codes_for_values(Series(v).values,
+                                                self.categories)
 
-        return self.codes.searchsorted(values_as_codes, sorter=sorter)
+        if -1 in values_as_codes:
```

Comment:
Isn't searching for 0's in the result of searchsorted() generally the same as searching for -1 among codes to be inserted?